test/e2e/volume_provisioning.go

```diff
@@ -0,0 +1,197 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"time"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/resource"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	// Requested size of the volume
+	requestedSize = "1500Mi"
+	// Expected size of the volume is 2GiB, because all three supported cloud
+	// providers  allocate volumes in 1GiB chunks.
+	expectedSize = "2Gi"
+)
+
+var _ = Describe("Dynamic provisioning of persistent volumes ", func() {
+	framework := NewFramework("volume-provisioning")
+
+	// filled in BeforeEach
+	var c *client.Client
+	var ns string
+
+	BeforeEach(func() {
+		c = framework.Client
+		ns = framework.Namespace.Name
+	})
+
+	////////////////////////////////////////////////////////////////////////
+	// Cinder
+	////////////////////////////////////////////////////////////////////////
+
+	Describe("DynamicProvisioner", func() {
+		It("should create and delete persistent volumes", func() {
+			SkipUnlessProviderIs("openstack", "gce", "aws")
+			By("creating a claim with a dynamic provisioning annotation")
+			claim := createClaim(ns)
+			defer func() {
+				c.PersistentVolumeClaims(ns).Delete(claim.Name)
+			}()
+			claim, err := c.PersistentVolumeClaims(ns).Create(claim)
+			Expect(err).NotTo(HaveOccurred())
+
+			err = waitForPersistentVolumeClaimPhase(api.ClaimBound, c, ns, claim.Name, 1*time.Second, 30*time.Second)
```

Comment:
In my experience, creation was always almost instant operation. Attachment was much slower. Still, we may hit some rate limiter in AWS, I'll increase it to 5 minutes.